lib/config/validation.spec.ts

```diff
@@ -1076,4 +1057,244 @@ describe('config/validation', () => {
+              'Invalid value `invalid` for `gitUrl`. The allowed values are default, ssh, endpoint.',
+            topic: 'Configuration Error',
+          },
+        ]);
+      });
+    });
+
+    it('validates boolean type options', async () => {
+      const config = {
+        unicodeEmoji: false,
+        detectGlobalManagerConfig: 'invalid-type',
+      };
+      const { warnings } = await configValidation.validateConfig(true, config);
+      expect(warnings).toMatchObject([
+        {
+          message: `Configuration option \`detectGlobalManagerConfig\` should be a boolean. Found: ${JSON.stringify(
+            'invalid-type',
+          )} (string).`,
+          topic: 'Configuration Error',
+        },
+      ]);
+    });
+
+    it('validates integer type options', async () => {
+      const config = {
+        prCommitsPerRunLimit: 2,
+        gitTimeout: 'invalid-type',
+      };
+      const { warnings } = await configValidation.validateConfig(true, config);
+      expect(warnings).toMatchObject([
+        {
+          message: `Configuration option \`gitTimeout\` should be an integer. Found: ${JSON.stringify(
+            'invalid-type',
+          )} (string).`,
+          topic: 'Configuration Error',
+        },
+      ]);
+    });
+
+    it('validates array type options', async () => {
+      const config = {
+        allowedPostUpgradeCommands: ['cmd'],
+        checkedBranches: 'invalid-type',
+        gitNoVerify: ['invalid'],
+      };
+      const { warnings } = await configValidation.validateConfig(
+        true,
+        // @ts-expect-error: contains invalid values
+        config,
+      );
+      expect(warnings).toMatchObject([
+        {
+          message:
+            'Configuration option `checkedBranches` should be a list (Array).',
+          topic: 'Configuration Error',
+        },
+        {
+          message:
+            'Invalid value for `gitNoVerify`. The allowed values are commit, push.',
+          topic: 'Configuration Error',
+        },
+      ]);
+    });
+
+    it('validates object type options', async () => {
+      const config = {
+        productLinks: {
+          documentation: 'https://docs.renovatebot.com/',
+          help: 'https://github.com/renovatebot/renovate/discussions',
+          homepage: 'https://github.com/renovatebot/renovate',
+        },
+        secrets: 'invalid-type',
+        cacheTtlOverride: {
+          someField: false,
+        },
+      };
+      const { warnings } = await configValidation.validateConfig(
+        true,
+        // @ts-expect-error: contains invalid values
+        config,
+      );
+      expect(warnings).toMatchObject([
+        {
+          message: 'Configuration option `secrets` should be a JSON object.',
+          topic: 'Configuration Error',
+        },
+        {
+          topic: 'Configuration Error',
+          message:
+            'Invalid `cacheTtlOverride.someField` configuration: value is not an integer.',
```

Comment:
```suggestion
            'Invalid `cacheTtlOverride.someField` configuration: value must be an integer.',
```

It seems nice to say what the expected/valid input is, instead of the thing that's broken. I'll let the maintainers decide what's best. :wink: